ansible_base/jwt_consumer/common/auth.py

```diff
@@ -258,17 +260,90 @@ def get_role_definition(self, name: str) -> Optional[Model]:
+        if not jwt_claims_hash:
+            logger.error("No claims_hash found in JWT token")
+            return
+
+        user_ansible_id = self.token.get("sub")
+        if not user_ansible_id:
+            logger.error("No subject (sub) found in JWT token")
+            return
+
+        # Check cached claims hash
+        cached_claims_hash = self.cache.get_cached_claims_hash(user_ansible_id)
+
+        if cached_claims_hash == jwt_claims_hash:
+            logger.debug(f"Claims hash matches cached value for user {user_ansible_id}")
+            return
+
+        # Calculate local claims hash
+        local_claims = get_user_claims(self.user)
+        local_hashable_claims = get_user_claims_hashable_form(local_claims)
+        local_claims_hash = get_claims_hash(local_hashable_claims)
+
+        if local_claims_hash == jwt_claims_hash:
+            logger.debug(f"Claims hash matches local calculation for user {user_ansible_id}")
+            # Update cache with the correct hash
+            self.cache.cache_claims_hash(user_ansible_id, jwt_claims_hash)
+            return
+
+        # Claims hash mismatch - fetch from gateway
+        logger.info(f"Claims hash mismatch for user {user_ansible_id}. JWT: {jwt_claims_hash}, Local: {local_claims_hash}. Fetching from gateway.")
+        gateway_claims = self._fetch_jwt_claims_from_gateway(user_ansible_id)
+
+        if gateway_claims:
+            # Extract claims structure from gateway response
+            objects = gateway_claims.get('objects', {})
+            object_roles = gateway_claims.get('object_roles', {})
+            global_roles = gateway_claims.get('global_roles', [])
+
+            # Process the RBAC permissions with the gateway claims
+            self._apply_rbac_permissions(objects, object_roles, global_roles)
+
+            # Update cache with the new hash
+            self.cache.cache_claims_hash(user_ansible_id, jwt_claims_hash)
+        else:
+            logger.error(f"Failed to fetch claims from gateway for user {user_ansible_id}")
+
+    def _fetch_jwt_claims_from_gateway(self, user_ansible_id: str) -> Optional[dict]:
+        """
+        Fetch JWT claims from the gateway endpoint using resource server client
+        """
+        try:
+            # Use the resource server client to make the request
+            client = get_resource_server_client(service_path="api/gateway/v1")
+
+            logger.debug(f"Fetching claims from gateway for user {user_ansible_id}")
+            response = client._make_request("GET", f"jwt_claims/{user_ansible_id}")
+
+            if response.status_code == 200:
+                claims_data = response.json()
+                logger.info(f"Successfully fetched claims from gateway for user {user_ansible_id}")
+                return claims_data
+            else:
+                logger.error(f"Gateway request failed with status {response.status_code}: {response.text}")
+                return None
+
+        except Exception as e:
```

Comment:
Pretty broad exception catching. Things I _think_ can fail:

`response = client._make_request("GET", f"jwt_claims/{user_ansible_id}")`

`claims_data = response.json()`

Reply:
You can dismiss this or address it, don't feel strongly.